src/kOS/Screen/TermWindow.cs

```diff
@@ -836,24 +842,21 @@ void TerminalGui(int windowId)
             keyClickEnabled = GUI.Toggle(keyClickButtonRect, keyClickEnabled, "Keyclicker", tinyToggleStyle);
             screen.Brightness = (double) GUI.VerticalSlider(brightnessRect, (float)screen.Brightness, 1f, 0f);
             GUI.DrawTexture(brightnessButtonRect, brightnessButtonImage);
-            
-            int charHeight = screen.CharacterPixelHeight;
 
-            CharacterInfo chInfo;
-            terminalLetterSkin.label.font.RequestCharactersInTexture("X"); // Make sure the char in the font is lazy-loaded by Unity.
-            terminalLetterSkin.label.font.GetCharacterInfo('X', out chInfo);
-
-            int charWidth = chInfo.advance;
+            int charHeight = screen.CharacterPixelHeight;
+            int charWidth = screen.CharacterPixelWidth;
 
+            // Note, pressing these buttons causes a change *next* OnGUI, not on this pass.
+            // Changing it in the midst of this pass confuses the terminal to change
+            // it's mind about how wide the window should be halfway through painting the
+            // components that make it up:
```

Comment:
A side effect of moving the font size calculations was that when you clicked the font resize buttons, the new values were getting ignored and overwritten so the buttons no longer did anything.  To make them work I had to postpone their effect until this redraw pass is finished, and make them take effect next time.